Assets/Scripts/EOSAchievementManager.cs

```diff
@@ -275,22 +184,61 @@ private static AchievementsInterface GetEOSAchievementInterface()
         /// Invoked when the request from the server has completed - whether
         /// successful or not.
         /// </param>
-        private static void QueryAchievements(ProductUserId productUserId, OnQueryDefinitionsCompleteCallback callback)
+        private Task<List<DefinitionV2>> QueryAchievementsAsync(ProductUserId productUserId)
```

Comment:
Should this have the "async" decorator? Maybe not, because this is managed through setting a task completion source?

Reply:
It should not have an `async` decorator, because nothing is `await`ed within the function. The function _itself_ is awaitable.

Reply:
I was under the impression that all awaitable things needed to be asynchronous, but doing some reading it appears as though it is unnecessary in some situations, including this situation here. So good to go.